hledger-lib/Hledger/Data/PeriodData.hs

```diff
@@ -73,18 +78,29 @@ instance Monoid a => Monoid (PeriodData a) where
 
 -- | Construct an 'PeriodData' from a list.
 periodDataFromList :: a -> [(Day, a)] -> PeriodData a
-periodDataFromList h = PeriodData h . IM.fromList . map (\(d, a) -> (fromInteger $ toModifiedJulianDay d, a))
+periodDataFromList h = PeriodData h . IM.fromList . map (\(d, a) -> (dayToInt d, a))
+
+-- | Convert 'PeriodData' to a list of pairs.
+periodDataToList :: PeriodData a -> (a, [(Day, a)])
+periodDataToList (PeriodData h as) = (h, map (\(s, e) -> (intToDay s, e)) $ IM.toList as)
 
--- | Get account balance information to the period containing a given 'Day'.
+
+-- | Get account balance information to the period containing a given 'Day',
+-- and the historical data if this day lies in the historical period.
 lookupPeriodData :: Day -> PeriodData a -> a
-lookupPeriodData d (PeriodData h as) =
-    maybe h snd $ IM.lookupLE (fromInteger $ toModifiedJulianDay d) as
+lookupPeriodData d pd@(PeriodData h _) = maybe h snd $ lookupPeriodDataLE d pd
+
+-- | Get account balance information for the period containing a given 'Day',
+-- along with the start of the period, or 'Nothing' if this day lies in the
+-- historical period.
+lookupPeriodDataLE :: Day -> PeriodData a -> Maybe (Day, a)
```

Comment:
What's LE ?

Reply:
Less than or equal to, named in parallel with the `lookupLE` function in IntMap. I agree it could be better named. Perhaps this could be `lookupPeriodData` and the previous `lookupPeriodData` could be `lookupPeriodDataOrHistorical`? I'm open to suggestions.

Reply:
Those sound good.